src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,467 @@
+	private bool ResetZoomAndOffsetWhenInactive { get; set; } = true;
+
+	public Size AvailableSize
+	{
+		get
+		{
+			var vOffset = (AdditionalMargin.Top + AdditionalMargin.Bottom);
+			var hOffset = (AdditionalMargin.Left + AdditionalMargin.Right);
+			return new Size(ActualWidth - hOffset, ActualHeight - vOffset);
+		}
+	}
+
+	// Methods
+	private async Task RaiseRenderedContentUpdated()
+	{
+		await Task.Yield();
+		RenderedContentUpdated?.Invoke(this, EventArgs.Empty);
+	}
+
+	private async void OnVerticalOffsetChanged()
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void OnHorizontalOffsetChanged()
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		UpdateHorizontalScrollBarValue();
+	}
+
+	private async void OnZoomLevelChanged()
+	{
+		CoerceZoomLevel();
+		UpdateScrollLimits();
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void UpdateContentBoundsVisibility()
+	{
+		if (_presenter?.Content is FrameworkElement fe)
+		{
+			var m = GetPositionMatrix(fe, this);
+
+			var flags = None;
+			if (m.OffsetX >= 0) flags |= BoundsVisibilityFlag.Left;
+			if (m.OffsetY >= 0) flags |= BoundsVisibilityFlag.Top;
+			if (ActualWidth >= (fe.ActualWidth * ZoomLevel) + m.OffsetX) flags |= BoundsVisibilityFlag.Right;
+			if (ActualHeight >= (fe.ActualHeight * ZoomLevel) + m.OffsetY) flags |= BoundsVisibilityFlag.Bottom;
+
+			ContentBoundsVisibility = flags;
+		}
+	}
+
+	private void UpdateScrollVisibility()
+	{
+		IsHorizontalScrollBarVisible = !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Left | BoundsVisibilityFlag.Right);
+		IsVerticalScrollBarVisible = !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Top | BoundsVisibilityFlag.Bottom);
+	}
+
+	private bool CanMoveIn((bool Horizontal, bool Vertical) _movementDirection)
+	{
+		if (ContentBoundsVisibility.HasFlag(All))
+		{
+			return false;
+		}
+
+		var canMove = false;
+		canMove |= CanScrollLeft() && _movementDirection.Horizontal is true;
+		canMove |= CanScrollRight() && _movementDirection.Horizontal is false;
+		canMove |= CanScrollUp() && _movementDirection.Vertical is true;
+		canMove |= CanScrollDown() && _movementDirection.Vertical is false;
+
+		return canMove;
+	}
+
+	private bool CanScrollUp() => !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Top);
+	private bool CanScrollDown() => !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Bottom);
+	private bool CanScrollLeft() => !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Left);
+	private bool CanScrollRight() => !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Right);
+
+	private async void UpdateHorizontalScrollBarValue()
+	{
+		HorizontalScrollValue = -1 * HorizontalOffset;
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void IsActiveChanged()
+	{
+		if (ResetZoomAndOffsetWhenInactive && !IsActive)
+		{
+			RemoveOffset();
```

Comment:
Why is this `RemoveOffset` and not `ResetOffset`?

Reply:
`RemoveOffset` resets the offsets to 0
`ResetOffset` resets the offsets to their initial values

When the control is no longer active, the intention is to remove any applied offsets i.e. reset the offsets entirely to zero
If we use ResetOffset() instead, we would potentially reapply margins, which might not be desired when deactivating the control